packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -147,6 +155,31 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
           }
         </div>
         <FocusZone>
+          {
+            showWeekNumbers ?
+              <table
+                className={ css('ms-DatePicker-weekNumbers', styles.weekNumbers, 'ms-DatePicker-table', styles.table) }
+              >
+                <tbody>
+                  { weekNumbers!.map((weekNumber, index) =>
+                    <tr key={ index }>
+                      <td>
+                        <div className={ css(
+                          'ms-DatePicker-day',
+                          styles.day,
+                          {
+                            ['ms-DatePicker-week--highlighted ' + styles.weekIsHighlighted]: (getDayOfYear(selectedDate) > (weekNumber - 1) * DAYS_IN_WEEK) && (getDayOfYear(selectedDate) < weekNumber * DAYS_IN_WEEK)
```

Comment:
getDayOfYear(selectedDate) is calculated twice in each iteration of the loop, you should move this outside of the loop and do it only once.